source/renderer/app/components/wallet/settings/ICOPublicKeyDialog.js

```diff
@@ -0,0 +1,138 @@
+// @flow
+import React, { Component } from 'react';
+import { observer } from 'mobx-react';
+import { defineMessages, intlShape } from 'react-intl';
+import { Input } from 'react-polymorph/lib/components/Input';
+import vjf from 'mobx-react-form/lib/validators/VJF';
+import styles from './ICOPublicKeyDialog.scss';
+import ReactToolboxMobxForm from '../../../utils/ReactToolboxMobxForm';
+import { submitOnEnter } from '../../../utils/form';
+import globalMessages from '../../../i18n/global-messages';
+import { FORM_VALIDATION_DEBOUNCE_WAIT } from '../../../config/timingConfig';
+import Dialog from '../../widgets/Dialog';
+import DialogCloseButton from '../../widgets/DialogCloseButton';
+import LocalizableError from '../../../i18n/LocalizableError';
+import { isValidSpendingPassword } from '../../../utils/validations';
+
+const messages = defineMessages({
+  title: {
+    id: 'wallet.settings.icoPublicKeyDialog.title',
+    defaultMessage: '!!!Reveal ICO public key',
+    description: 'Title "Choose a stake pool" on the reveal Wallet Id dialog.',
+  },
+  description: {
+    id: 'wallet.settings.icoPublicKeyDialog.description',
+    defaultMessage:
+      '!!!Please enter your spending password to reveal your ICO’s public key.',
+    description: 'Description on the reveal Wallet Id dialog.',
+  },
+  buttonLabel: {
+    id: 'wallet.settings.icoPublicKeyDialog.button',
+    defaultMessage: '!!!Reveal ICO public key',
+    description: 'Description on the reveal ICO Id dialog.',
+  },
+});
+
+type Props = {
+  onRevealPublicKey: Function,
+  onClose: Function,
+  error: ?LocalizableError,
+  hasReceivedICOPublicKey: boolean,
+};
+
+@observer
+export default class ICOPublicKeyDialog extends Component<Props> {
```

Comment:
Let's also use a functional component here?

Reply:
I tried really but I couldn't make it working with MobxReactForm